### PR TITLE
Sample status

### PR DESCRIPTION
**Pull request by akl21**
I am trying to create a pull request so you can access my updated data_validation.R file, which is now uploaded to my sample_status branch.

### CODE REVIEW

**Comment by fmichonneau** [comment]
It's working: I got your pull request. I'll review the code tomorrow.

**Comment by fmichonneau** [comment]
Hi @akl21,

  Thank you that looks good!

  Because this code is being integrated into a package there a few conventions that you need to follow. One of them is that there shouldn't be anything but functions in the code, so please remove the beginning (the calls to `read.csv`) and the calls to the individual functions. I'll also tag them in the file.

   I need to improve how the data is being loaded in the package so you can easily have access to it, but for now I'll just let you know what you need to do to call the correct data with the functions.

  Once you make these modifications in your copy of the code, you can just push to your branch and the changes will automatically be brought in this pull request. Leave a comment when you are done so I can review and integrate your modifications.

File: R/data_validation.R [hunk 1]
```diff
@@ -0,0 +1,57 @@
+pcr_table <- read.csv(file = "R/CopyOfdata/pcr_data.csv",
```
Comment (fmichonneau) [inline R/data_validation.R]: please remove l.1-8

File: R/data_validation.R [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+phylum_table <- read.csv("R/CopyOfdata/phylum_listings.csv", 
+                         stringsAsFactors= FALSE)
+extraction_table <- read.csv(file = "R/CopyOfdata/extraction_data.csv",
+                             stringsAsFactors = FALSE)
+sequencing_plate_table <- read.csv(file = "R/CopyOfdata/sequencing_plate_data.csv",
+                                   stringsAsFactors = FALSE)
+
+#This function checks if the voucher numbers in the PCR data, extraction data
+# and the sequencing plate data are in the phylum listings.
+voucher_num_present <- function(pcr_tbl= pcr_table,
```
Comment (fmichonneau) [inline R/data_validation.R]: `pcr_tbl = get_lab("pcr_sample_data")`

File: R/data_validation.R [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                                phylum_tbl= phylum_table,
```
Comment (fmichonneau) [inline R/data_validation.R]: `phylum_tbl = get_lab("sample_data")`

File: R/data_validation.R [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+extraction_table <- read.csv(file = "R/CopyOfdata/extraction_data.csv",
+                             stringsAsFactors = FALSE)
+sequencing_plate_table <- read.csv(file = "R/CopyOfdata/sequencing_plate_data.csv",
+                                   stringsAsFactors = FALSE)
+
+#This function checks if the voucher numbers in the PCR data, extraction data
+# and the sequencing plate data are in the phylum listings.
+voucher_num_present <- function(pcr_tbl= pcr_table,
+                                phylum_tbl= phylum_table, 
+                                ext_tbl = extraction_table,
```
Comment (fmichonneau) [inline R/data_validation.R]: `ext_tbl = get_lab("extraction_data")`

File: R/data_validation.R [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                             stringsAsFactors = FALSE)
+sequencing_plate_table <- read.csv(file = "R/CopyOfdata/sequencing_plate_data.csv",
+                                   stringsAsFactors = FALSE)
+
+#This function checks if the voucher numbers in the PCR data, extraction data
+# and the sequencing plate data are in the phylum listings.
+voucher_num_present <- function(pcr_tbl= pcr_table,
+                                phylum_tbl= phylum_table, 
+                                ext_tbl = extraction_table, 
+                                seq_plate_tbl= sequencing_plate_table){
```
Comment (fmichonneau) [inline R/data_validation.R]: `seq_plate_tbl = get_lab("sequencing_plate_data")`

File: R/data_validation.R [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    not_in_ext <-  ext_tbl$voucher_number[!in_ext]
+    warning("Some vouchers in extractions not in phylum listings:", paste(not_in_ext, collapse= ", "))
+  }
+  if (!all(in_seq_plate)){
+    not_in_seq_plate <-  in_seq_plate$voucher_number[!in_seq_plate]
+    warning("Some vouchers in sequencing plate data not in phylum listings:",
+           paste(not_in_seq_plate, collapse= ", "))
+  }
+}
+voucher_num_present()
```
Comment (fmichonneau) [inline R/data_validation.R]: please also remove the calls to the functions

File: R/data_validation.R [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+#This function checks that there are no duplicate voucher numbers in the phylum listings.
+no_duplicates <- function(phylum_tbl = phylum_table){
```
Comment (fmichonneau) [inline R/data_validation.R]: `phylum_tbl = get_lab("sample_data")`

File: R/data_validation.R [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    stop("There are duplicate voucher numbers in the phylum listings", duplicates)
+  }
+}
+no_duplicates() 
+
+
+#This function attempts to check that each combination of pcr_id and 
+#voucher_number in the sequencing plate data is also found in the 
+#PCR data. 
+comb_present = function(seq_plate_tbl = sequencing_plate_table,
```
Comment (fmichonneau) [inline R/data_validation.R]: seq_plate_tbl = get_lab("sequencing_plate_data")
pcr_tbl = get_lab("pcr_sample_data")

**Comment by akl21** [comment]
I finished writing the function to produce a list of data for the voucher number entered, named get_info_voucher.R. It is uploaded to the sample_status branch.

**Pull request closed without merging**